Atomic/pull.py

```diff
@@ -29,5 +31,75 @@ def pull_image(self):
 
+    def discover_sigstore(self):
+        """
+        Check for registry/repo/sigstore metadata image
+        prompt user for trust on first use workflow
+        :return: True if sigstore discovered and configured
+        """
+        (registry, repo, _) = decompose(self.args.image)
+        # FIXME: this should be handled in util.decompose
+        _repo, _image = repo.split('/')
+        # TODO: check local /etc/containers/registries.d config here
+        repo_sigstore_labels = self._get_sigstore_image_metadata(registry, _repo)
+        if self._validate_sigstore_labels(repo_sigstore_labels):
+            if self._prompt_trust(repo_sigstore_labels):
+                discover_config = False
+                trust_scope = "%s/%s" % (registry, _repo)
+                discover_config = write_registry_config(trust_scope)
+                pubkey_path = install_pubkey(repo_sigstore_labels['pubkey-id'], repo_sigstore_labels['pubkey-url'])
+                discover_config = update_trust_policy(trust_scope, pubkey_path, repo_sigstore_labels['sigstore-url'])
+                return discover_config
+        return False
+
+    def _get_sigstore_image_metadata(self, registry, repo):
+        """
+        Get sigstore metadata image
+        :param registry: registry string
+        :param repo: repo string
+        :return True on success
+        """
+        _img = get_atomic_config_item(['sigstore_metadata_image'], get_atomic_config())
+        sigstoreimage = '/'.join([registry, repo, _img])
+        data = skopeo_inspect("docker://" + sigstoreimage, args=None, fail_silent=True)
+        if data:
+            write_out("Found registry sigstore metadata image %s" % sigstoreimage)
+            return data['Labels']
+        else:
+            return False
+
+    def _validate_sigstore_labels(self, labels):
+        """
+        Validate sigstore metadata.
+        If there's a missing key or something we don't want to perform any automatic trust policy configuration
+        :param labels: unvalidated labels. Should be either dict or False
+        :return: True if labels are valid
+        """
+        valid = False
+        if labels:
+            expected_keys = ["pubkey-id", "pubkey-fingerprint", "pubkey-url", "sigstore-url"]
+            for k in expected_keys:
+                valid = k in labels
+        return valid
+
+    def _prompt_trust(self, labels):
+        """
+        Prompt user for trust on first use workflow
+        :param labels: dict of metadata labels defining sigstore trust
+        :return: True if user accepts
+        """
+        write_out("ID: " + labels['pubkey-id'])
+        write_out("Fingerprint: " + labels['pubkey-fingerprint'])
+        write_out("Public key download URL: %s" % labels['pubkey-url'])
+        if self.args.trust:
+            return True
+        trust = raw_input("Do you want to add trust policy for this registry? (y/N) ")
```

Comment:
@runcom `raw_input()` is failing python 3. I wasn't sure about our compatibility goals. Should I use `input()`?